src/types/Cache.ts

```diff
@@ -0,0 +1,15 @@
+import { SecondLevelDomain } from "./SecondLevelDomain";
+import { TopLevelDomain } from "./TopLevelDomain";
+
+export interface ICache {
+  getTld(tldHash: string): Promise<TopLevelDomain> | TopLevelDomain | undefined;
+  getTlds(): Promise<TopLevelDomain[]> | TopLevelDomain[] | undefined;
+  getSld(
+    tldHash: string,
+    sldHash: string
+  ): Promise<SecondLevelDomain> | SecondLevelDomain | undefined;
+  setTld(tldHash: string, tld: TopLevelDomain): void | any;
```

Comment:
`Promise<void>` may be a better return type. Non-async caches can return `Promise.resolve()` to satisfy this constraint. Network caching should be first class over in-memory or other sync caches.